### PR TITLE
Quadhoral openmeteo alex

### LINKED ISSUE #17: OpenMeteo Quadhoral Pipeline transform.py
As an amateur stargazer, I want the weather data to be cleaned, filtered, and transformed in a meaningful way, such as converting temperature units or filtering cloud cover percentage, so I can quickly assess the data without unnecessary details.

- Transform nested list to either `pandas` dataframe or list of list

### PR DESCRIPTION
**Pull request by AlexOBarnes**
This closes #17 #22 

What I have done:
- Formatted code
- Written tests for `transform.py`
- Implemented `transform.py` functions

### CODE REVIEW

File: quadhoral-openmeteo/conftest.py [hunk 1]
```diff
@@ -16,7 +17,7 @@ def clean_valid_coord():
 @pytest.fixture
 def valid_json_data():
     '''Returns an example data from API'''
-    return [{'hourly': {'temperature_2m': [15], 
+    return [{'hourly': {'temperature_2m': [15],
```
Comment (CalebMathers) [inline quadhoral-openmeteo/conftest.py:20]: Cya later whitespace, I hope you're happy pylint

File: quadhoral-openmeteo/conftest.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 '''Contains the configuration for the tests in this file'''
+from datetime import datetime as dt
```
Comment (MurasatoNaoya) [inline quadhoral-openmeteo/conftest.py:2]: A wide selection of fixtures covering the wide variety of weather data, LGTM.

File: quadhoral-openmeteo/conftest.py (hunk 1 above)
Comment (olliejth) [reply quadhoral-openmeteo/conftest.py:20]: Enjoy the money pylint.

File: quadhoral-openmeteo/quadhoral_transform.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        raise ValueError('No data to transform.')
+
+    formatted_data = []
+    logging.info('Data transformation begun.')
+
+    for county_id, county in enumerate(data, 1):
+        logging.info('Cleaning data for county: %s.', county_id)
+        clean_county = clean_data(county)
+        formatted_data.extend(format_data(clean_county, county_id))
+        logging.info('Data cleaning for county %s complete.', county_id)
```
Comment (CalebMathers) [inline quadhoral-openmeteo/quadhoral_transform.py:41]: Good logging

File: quadhoral-openmeteo/conftest.py [hunk 4]
```diff
@@ -26,6 +27,71 @@ def valid_json_data():
...
+    '''Returns valid clean county data'''
+    return {
+        'time': [f'2024-10-01T{i:02d}:00' for i in range(24)],
+        'temp': [10 + i for i in range(24)],
+        'prob': [20 + i for i in range(24)],
+        'precip': [0.1 * i for i in range(24)],
+        'clouds': [50 + i for i in range(24)],
+        'visibility': [10000 - (i * 100) for i in range(24)]
+    }
+
```
Comment (olliejth) [inline quadhoral-openmeteo/conftest.py:45]: Interesting fixture

File: quadhoral-openmeteo/conftest.py [hunk 5]
```diff
@@ -26,6 +27,71 @@ def valid_json_data():
...
+def valid_clean_county_data():
+    '''Returns valid clean county data'''
+    return {
+        'time': [f'2024-10-01T{i:02d}:00' for i in range(24)],
+        'temp': [10 + i for i in range(24)],
+        'prob': [20 + i for i in range(24)],
+        'precip': [0.1 * i for i in range(24)],
+        'clouds': [50 + i for i in range(24)],
+        'visibility': [10000 - (i * 100) for i in range(24)]
+    }
```
Comment (CalebMathers) [inline quadhoral-openmeteo/conftest.py:44]: Good use of fixtures

File: quadhoral-openmeteo/test_quadhoral_transform.py [hunk 6]
```diff
@@ -0,0 +1,86 @@
+'''Contains tests for transform script'''
```
Comment (MurasatoNaoya) [inline quadhoral-openmeteo/test_quadhoral_transform.py:1]: Nice carry over from other pipelines! Zoo wee mama!

**Review by MurasatoNaoya — APPROVED** [review]
LGTM.

File: quadhoral-openmeteo/quadhoral_transform.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+             clean_county_data['precip'][i], clean_county_data['clouds'][i],
+             clean_county_data['visibility'][i]] for i in range(24)]
+
+def transform(data: list[dict]) -> list[list]:
+    '''Orchestrates transform steps of pipeline'''
+    if not data:
+        raise ValueError('No data to transform.')
+
+    formatted_data = []
+    logging.info('Data transformation begun.')
```
Comment (MurasatoNaoya) [inline quadhoral-openmeteo/quadhoral_transform.py:35]: Clear logging that provides records of significant steps of the transform script.

**Review by MurasatoNaoya — APPROVED** [review]

**Review by CalebMathers — APPROVED** [review]

File: quadhoral-openmeteo/quadhoral_extract.py [hunk 8]
```diff
@@ -69,4 +69,4 @@ def extract() -> list:
     logging.basicConfig(level=logging.INFO)
     load_dotenv()
     logging.info('Environment loaded')
```
Comment (olliejth) [inline quadhoral-openmeteo/quadhoral_extract.py:71]: Nice use of print to debug.

**Review by olliejth — APPROVED** [review]
Overall I see no issues, may need to be linted later.

**Pull request merged**